sdk/core/azure-core/inc/context.hpp

```diff
@@ -144,19 +144,30 @@ namespace Azure { namespace Core {
   private:
     struct ContextSharedState
     {
-      std::shared_ptr<ContextSharedState> Parent;
-      time_point CancelAt;
-      std::string Key;
-      ContextValue Value;
+      std::shared_ptr<ContextSharedState> const Parent;
+      std::chrono::milliseconds CancelAtMsecSinceEpoch;
+      std::string const Key;
+      ContextValue const Value;
```

Comment:
@BillyONeal , I am purposefully making `const` data members here (except for `CancelAtMsecSinceEpoch`) - this way, 
I am now sure that they are multithread-safe, and noone is trying to modify them. If anyone would try to modify them I would actually like for that code to not compile - so that you'll have to add mutex and only then remove const.

Reply:
Storing the point in time rather than as milliseconds reduces the number of conversations that must be done

Reply:
Switch from time_point to int64 milliseconds is intentional to achieve thread safety.

Reply:
@BillyONeal / @CaseyCarter : Is volatile on the field good enough or do we need some kind of interlocked read/write when reading/writing to the field?

Reply:
64-bit values are not atomic updates on 32-bit OSs

Reply:
You should use `std::atomic<the type>` if you want portability. `volatile` means *nothing* for purposes of multithreading.

Reply:
Sorry, I didn't actually look at the code before. I don't understand the motivation for this change. The constness of the fields are achieved by the pointer to the bits being pointer-to-`const` so the fields need not be const.